### DISCUSSION TITLE
Adding Multiple API Headers for Authentication in Dify Custom Tools

### DISCUSSION THREAD

**Discussion by thejokertm** [langgenius/dify#6739] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Hello,

I’m currently trying to configure a custom API tool in Dify for Alpaca and need some help with authentication. Alpaca requires both an API key and a secret to be included in the headers.

Is there a way to configure multiple headers for authentication in Dify for the custom API tools?

I could use the HTTP request node in the workflow settings, but this is very cumbersome, as for every action I need to basically create a new workflow, since you can’t have multiple tools or actions in one workflow, at least that is my assumption. 

I have looked through the documentation but couldn’t find a clear solution. Any guidance on how to set this up would be greatly appreciated!

Thanks in advance.

**Comment by maxi-speed** [comment]
i have the same issue

**Comment by seta-namdh-7131** [comment]
Same issue here, has anyone found the solution?

**Comment by brettgoodrich** [comment]
Same issue!
